pkg/processor/runtime/rpc/abstract.go

```diff
@@ -150,6 +151,24 @@ func (r *AbstractRuntime) Stop() error {
 
 	if r.wrapperProcess != nil {
 
+		if r.configuration.ExplicitAckEnabled {
+
+			// signal wrapper to terminate
+			r.Logger.DebugWith("Sending termination signal to wrapper process",
+				"wrapperProcess", r.wrapperProcess)
+
+			if err := r.wrapperProcess.Signal(syscall.SIGTERM); err != nil {
+				r.Logger.WarnWith("Failed to signal termination to wrapper process")
+				return errors.Wrap(err, "Can't signal termination to wrapper process")
+			}
+
+			// wait for workers to finish
+			r.Logger.DebugWith("Waiting for wrapper process to stop processing events",
+				"wrapperProcess", r.wrapperProcess,
+				"waitTime", r.configuration.WorkerTerminationTimeout)
+			time.Sleep(r.configuration.WorkerTerminationTimeout)
```

Comment:
The idea is to let the workers do "some stuff" before terminating the process.
Since "some stuff" is up to the user and can change, we don't wait on something specific.

With the way the communication between the processor and wrapper process works - If we do wait for an OK signal of some sort from the wrapper process, it will arrive on a channel to the trigger itself, and not to the underlying runtime. Sending it to the runtime again will result in a kind of loop.

Reply:
I think that @quaark  is right here. `WorkerTerminationTimeout` should be a hard stop but perhaps the runtime itself may "exit" to avoid wasting most of the time. thus, perhaps, you can sleep (loop/select) and verify the process is still "alive". if process is not alive, you may stop "sleeping" and move on